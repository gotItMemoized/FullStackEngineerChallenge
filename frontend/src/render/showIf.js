export default (statement, component) => {
  if (statement === true) {
    // function check voodoo
    if (component instanceof Function) {
      return component();
    }
    return component;
  } else if (statement !== false) {
    throw new Error('Expected a true or false statement');
  }
};
