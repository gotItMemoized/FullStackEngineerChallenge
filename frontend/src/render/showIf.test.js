import React from 'react';
import { render } from '@testing-library/react';
import showIf from './showIf';
import 'jest-dom/extend-expect';

it('should not render on false', () => {
  const { container } = render(showIf(false, <div id="test">hello</div>));
  expect(container).toMatchSnapshot();
});

it('should render on true', () => {
  const { container } = render(showIf(true, <div id="test">hello</div>));
  expect(container).toMatchSnapshot();
  const innerDiv = document.getElementById('test');
  expect(innerDiv).toMatchSnapshot();
  expect(innerDiv).toHaveTextContent('hello');
});

it('should explode on not true', () => {
  const renderComponent = () => render(showIf({}, <div id="test">hello</div>));
  expect(renderComponent).toThrow('Expected a true or false statement');
});

it('should execute function if sent in', () => {
  const result = showIf(true, () => 'hello');
  expect(result).toEqual('hello');
});
