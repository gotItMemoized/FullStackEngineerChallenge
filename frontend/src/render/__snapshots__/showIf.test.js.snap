// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render on false 1`] = `<div />`;

exports[`should render on true 1`] = `
<div>
  <div
    id="test"
  >
    hello
  </div>
</div>
`;

exports[`should render on true 2`] = `
<div
  id="test"
>
  hello
</div>
`;
